examples/local/local-container-mnist.ipynb

```diff
@@ -83,25 +85,62 @@
     "            x = self.fc2(x)\n",
     "            return F.log_softmax(x, dim=1)\n",
     "\n",
-    "    # Create the model\n",
-    "    model = Net()\n",
-    "    \n",
-    "    # Load MNIST dataset\n",
-    "    dataset = datasets.MNIST(\n",
-    "        './data',\n",
-    "        train=True,\n",
-    "        download=True,\n",
-    "        transform=transforms.Compose([transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))])\n",
+    "    # Use NCCL if a GPU is available, otherwise use Gloo as communication backend.\n",
+    "    device, backend = (\"cuda\", \"nccl\") if torch.cuda.is_available() else (\"cpu\", \"gloo\")\n",
+    "    print(f\"Using Device: {device}, Backend: {backend}\")\n",
+    "\n",
+    "    # Setup PyTorch distributed.\n",
+    "    local_rank = int(os.getenv(\"LOCAL_RANK\", 0))\n",
+    "    dist.init_process_group(backend=backend)\n",
+    "    print(\n",
+    "        \"Distributed Training for WORLD_SIZE: {}, RANK: {}, LOCAL_RANK: {}\".format(\n",
+    "            dist.get_world_size(),\n",
+    "            dist.get_rank(),\n",
+    "            local_rank,\n",
+    "        )\n",
     "    )\n",
-    "    train_loader = DataLoader(dataset, batch_size=64, shuffle=True)\n",
+    "\n",
+    "    # Create the model and load it into the device.\n",
+    "    device = torch.device(f\"{device}:{local_rank}\")\n",
+    "    model = nn.parallel.DistributedDataParallel(Net().to(device))\n",
     "    optimizer = optim.SGD(model.parameters(), lr=0.01, momentum=0.5)\n",
-    "    \n",
+    "\n",
+    "    # Download FashionMNIST dataset only on local_rank=0 process.\n",
+    "    if local_rank == 0:\n",
+    "        dataset = datasets.FashionMNIST(\n",
+    "            \"data\",\n",
+    "            train=True,\n",
+    "            download=True,\n",
+    "            transform=transforms.Compose([\n",
+    "                transforms.ToTensor(),\n",
+    "                transforms.Normalize((0.1307,), (0.3081,)),\n",
+    "            ]),\n",
+    "        )\n",
+    "        dist.barrier()\n",
+    "    else:\n",
+    "        dist.barrier()\n",
+    "        dataset = datasets.FashionMNIST(\n",
+    "            \"data\",\n",
+    "            train=True,\n",
+    "            download=False,\n",
+    "            transform=transforms.Compose([transforms.ToTensor()]),\n",
+    "        )\n",
```

Comment:
else is not needed since torch will just load dataset in the master node, check how we did it here: https://github.com/kubeflow/trainer/blob/master/examples/pytorch/image-classification/mnist.ipynb

Reply:
Ah you're right, updated!